sgl-kernel/src/sgl-kernel/csrc/quantization/fp4/nvfp4_scaled_mm_kernels.cu

```diff
@@ -0,0 +1,298 @@
+#include <ATen/cuda/CUDAContext.h>
+#include <c10/cuda/CUDAGuard.h>
+#include <torch/all.h>
+
+#include "cutlass/cutlass.h"
+#include "cutlass/epilogue/collective/collective_builder.hpp"
+#include "cutlass/gemm/collective/collective_builder.hpp"
+#include "cutlass/gemm/device/gemm_universal_adapter.h"
+#include "cutlass/gemm/kernel/gemm_universal.hpp"
+#include "cutlass/util/packed_stride.hpp"
+#include "cutlass_extensions/common.hpp"
+
+using namespace cute;
+
+#if defined(CUTLASS_ARCH_MMA_SM100_SUPPORTED)
+struct Fp4GemmSm100Float {
```

Comment:
maybe we can use template to reduce the dups. (like [here](https://github.com/vllm-project/vllm/blob/82e0d601fc9bc4cdbf307e06d3ed6faf9f4dd72e/csrc/quantization/fp4/nvfp4_scaled_mm_kernels.cu#L37-L59))

Reply:
Thanks, fixed